api/pkg/transformer/pipeline/testdata/valid_table_transform_conditional_filtering.yaml

```diff
@@ -0,0 +1,46 @@
+transformerConfig:
+  preprocess:
+    inputs:
+      - variables:
+          - name: customer_id
+            jsonPath: $.customer.id
+          - name: zero
+            literal:
+              intValue: 0
+      - tables:
+          - name: driver_table
+            baseTable:
+              fromJson:
+                jsonPath: $.drivers[*]
+    transformations:
+      - tableTransformation:
+          inputTable: driver_table
+          outputTable: transformed_driver_table
+          steps:
+            - updateColumns:
+              - column: customer_id
+                expression: customer_id
+              - column: driver_performa
+                conditions:
+                  - if: driver_table.Col("rating") * 2 <= 7
+                    expression: driver_table.Col("rating") * 1
```

Comment:
So this expression will only apply to rows that satisfy `driver_table.Col("rating") * 2 <= 7` condition, right? I now feel calling it `if` feel strange. Maybe `toRowsIf` sounds better?

Reply:
or `rowSelector` ?

Reply:
`rowSelector` sounds better

Reply:
sounds good. But `conditions` a bit not clear. Maybe call it `selectiveTransform` ?